docs/source/specs/openapi.json

```diff
@@ -1699,7 +1716,13 @@
               "data": {
                 "type": "array",
                 "items": {
-                  "$ref": "#/components/schemas/Access"
+                  "oneOf": [{
```

Comment:
@chambridge Since this API returns one of the 2 objects would it make sense to have a separate API get_principal_roles distinct from get_principal_access?

Reply:
I don't have a strong opinion either way. I think it works as is, since we are looking at the breakdown of access for a principal. Role is a higher level than the permissions they would see. However, if you feel like it would be best to separate the two, I'm open to suggestions for the API endpoint. The above currently lives off: `api/rbac/v1/access/`

The following already exist and perform other functions:
`api/rbac/v1/principals/`
`api/rbac/v1/groups/`
`api/rbac/v1/roles/`
`api/rbac/v1/policies/`

Reply:
@chambridge My preference would be a get_principal_roles in the RolesAPI class which would complement the get_principal_access in the AccessAPI class.

Reply:
Ok. What would the endpoint look? The OpenApi operationId doesn't really matter.

_GET_ `api/rbac/v1/roles/` - Returns all roles for the account, not for the calling user and any change to that functionality break the on-boarding flow and UI.

Could do something like:
_GET_ `api/rbac/v1/roles/current/` or _GET_ `api/rbac/v1/roles/me/` but those don't seem like an improvement to me over what is currently suggested.

Reply:
@chambridge How about api/rbac/v1/roles?scope=principal and the default is
api/rbac/v1/roles which uses the default scope=tenant and doesn't break backward compatibility

Reply:
ok. I'm going to close this. and I'll work thru the above.